pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/blocks/TransferableBlock.java

```diff
@@ -146,10 +146,18 @@ public DataBlock.Type getType() {
    *
    * @return whether this block is the end of stream.
    */
+  // TODO: Update the name to isTerminateBlock.
   public boolean isEndOfStreamBlock() {
     return isType(MetadataBlock.MetadataBlockType.ERROR) || isType(MetadataBlock.MetadataBlockType.EOS);
   }
 
+  /**
+   * @return true when the block is a real end of stream block instead of error block.
+   */
+  public boolean isEOSBlock() {
+    return isType(MetadataBlock.MetadataBlockType.EOS);
+  }
```

Comment:
IMO, simply renamed this to 
`isSuccessfulEndOfStreamBlock()` should be suffice. makes it less confusing and no need to a refactoring.

Reply:
Done